Velocity_Model/scripts/gen_basin_stats.py

```diff
@@ -0,0 +1,166 @@
+import datetime
+import itertools
+from pathlib import Path
+
+
+import matplotlib.path as mpltPath
+import numpy as np
+
+from qcore import geo
+
+from Velocity_Model.z import basin_dict
+from mpi4py import MPI
+
+
+x_start = 1060100
+x_end = 2120000
+y_start = 4730100
+y_end = 6250000
+spacing = 100
+x_range = range(x_start, x_end, spacing)
+y_range = range(y_start, y_end, spacing)
+
+comm = MPI.COMM_WORLD
+rank = comm.Get_rank()
+num_cores =  comm.Get_size()
+
+
+BASIN_VER = "2.07"
+basins = basin_dict[BASIN_VER]
+
+outdir = Path("outdir")
+out_csv = outdir / f"basin_stats_{rank:04d}.csv" # lon, lat, nztm_x, nztm_y, True, basin_name
+#168.2632171899503, -44.71216293699213, 1224800, 5038100, True, WakatipuBasinOutlineWGS84.txt
+
+out_ll = outdir / f"basin_stats_{rank:04d}.ll" # lon lat nztm_x_nztm_y
+#168.2632171899503 -44.71216293699213 1224800_5038100
+
+checkpoint = outdir / f"checkpoint_{rank:04d}.txt"
+
+class Buffer:
```

Comment:
The standard file i/o has buffering? https://docs.python.org/3/library/functions.html#open
Was this not configurable enough for this use case?

Reply:
This is a checkpoint/resume enabled buffer. If I only cared about staggered file I/O, I'd guess what you suggest is adequate, but this built-in function doesn't seem to give me enough control for checkpoint/resuming.

Reply:
Sweet. Can you update the comment to say what it's for. From the comment it looks like the main reason for it is the performance.